pkg/verification/verification_service.go

```
@@ -147,56 +189,78 @@ func generateVerificationCode() (string, error) {
 
 // VerifyCode validates the user's phone verification code.  It updates the specified UserSignup value, so even
 // if an error is returned by this function the caller should still process changes to it
-func (s *ServiceImpl) VerifyCode(signup *v1alpha1.UserSignup, code string) (*v1alpha1.UserSignup, error) {
+func (s *ServiceImpl) VerifyCode(ctx *gin.Context, userID string, code string) (verificationErr error) {
+
+	signup, verificationErr := s.Services().SignupService().GetUserSignup(userID)
+	if verificationErr != nil {
+		log.Error(ctx, verificationErr, "error retrieving usersignup")
+		return
```

is there a reason why the error is not being returned here?